src/silo/storage/dictionary.test.cpp

```diff
@@ -0,0 +1,82 @@
+#include "silo/storage/dictionary.h"
+
+#include <gtest/gtest.h>
+#include <fstream>
+
+#include "silo/storage/pango_lineage_alias.h"
+
+using silo::Dictionary;
+
+TEST(Dictionary, shouldBuildWithoutErrors) {
+   ASSERT_NO_THROW(Dictionary());
+}
+
+TEST(Dictionary, shouldHavePredefinedCoulmns) {
+   const Dictionary dictionary;
+
+   ASSERT_NO_THROW(dictionary.lookupValueId("country", "someCountry"));
+   ASSERT_THROW(dictionary.lookupValueId("noValidColumn", "someValue"), std::exception);
+}
+
+class DictionaryTestFixture : public ::testing::Test {
+  protected:
+   Dictionary dictionary;
+
+   void SetUp() override {
+      const std::string directory = "testBaseData/";
+
+      auto alias = silo::PangoLineageAliasLookup::readFromFile(directory);
+      dictionary.updateDictionary(directory + "small_metadata_set.tsv", alias);
+   }
+};
+
+TEST_F(DictionaryTestFixture, shouldUpdateDictionary) {
+   ASSERT_TRUE(dictionary.lookupValueId("country", "Switzerland").has_value());
+   ASSERT_TRUE(dictionary.lookupValueId("division", "Bern").has_value());
+}
+
+TEST_F(DictionaryTestFixture, lookupValueIdshouldReturnEmptyOptionalIfNotExists) {
+   ASSERT_FALSE(dictionary.lookupValueId("country", "NotACountry").has_value());
+}
+
+TEST_F(DictionaryTestFixture, shouldLookupValueId) {
+   ASSERT_EQ(dictionary.lookupValueId("country", "Switzerland"), 0);
+   ASSERT_EQ(dictionary.lookupValueId("division", "Bern"), 1);
+
+   ASSERT_EQ(
+      dictionary.lookupValueId("division", "Bern"), dictionary.lookupValueId("division", "Bern")
+   );
```

Comment:
Why do we need that assertion?^^

Reply:
I thought that reading a value twice should still result in the same value. But is not neccessary per se :)